### PR TITLE
[lldb] Update for removal of `ModuleDecl::addFile`

### PR DESCRIPTION
**Pull request by hamishknight**
Update for https://github.com/swiftlang/swift/pull/77666

### CODE REVIEW

**Comment by hamishknight** [comment]
https://github.com/swiftlang/swift/pull/77666

@swift-ci please test

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.h [hunk 1]
```diff
@@ -311,12 +311,20 @@ class SwiftASTContext : public TypeSystemSwift {
   /// \return the ExtraArgs of the ClangImporterOptions.
   const std::vector<std::string> &GetClangArguments();
 
+  /// Attempt to create an empty Swift module.
+  ///
+  /// \param importInfo Information about which modules should be implicitly
+  /// imported by each file of the module.
```
Comment (hamishknight) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.h]: Need to remove this

**Comment by hamishknight** [comment]
https://github.com/swiftlang/swift/pull/77666

@swift-ci please test

**Comment by hamishknight** [comment]
https://github.com/swiftlang/swift/pull/77666

@swift-ci please test

**Review by adrian-prantl — CHANGES_REQUESTED** [review]
Can you rebase this PR on swift/release/6.1?

**Comment by hamishknight** [comment]
@adrian-prantl This is targeting `main`, it doesn't need to be in 6.1

**Review by adrian-prantl — APPROVED** [review]
In that case LGTM — please don't forget to also create a matching PR that cherry-picks this to `next`!

**Comment by adrian-prantl** [comment]
You can/should merge the next cherry-pick immediately without review or testing.

**Pull request merged**